python-leap solution

cmdellinger opened the pull request:
solution to exercism.io leap exercise in python

cc @barberj

Commit:
python-leap solution

solution to exercise.io leap exercise

barberj commented on python/leap/leap.py:
```diff
@@ -0,0 +1,65 @@
+## ------
+
+## test whether an argument is passed in the command line
+if len(sys.argv) > 1:
```
its been a while since i've done python... but [docopt](https://github.com/docopt/docopt) is a cool library that i've used in the past for command line stuff. Refactor and try your hand at using a third party library.

barberj commented on python/leap/leap.py:
```diff
@@ -0,0 +1,65 @@
+    i = 1
+    
+    ## loop to test all arguments
+    while i < len(sys.argv):
```
Instead of iterating by index are you familiar with python [iterators](http://anandology.com/python-practice-book/iterators.html)?

barberj commented on python/leap/leap.py:
```diff
@@ -0,0 +1,65 @@
+        if year%100 == 0 and year%400 != 0:
+            return False
+        return True
+    return False
```
can you simplify by reducing the nesting of conditionals?

Commit:
added parser & refactored is_leap_year

Changes:
- Refactored is_leap_year() to remove nested conditions.
- Added docopt parser to replace manual parsing.
*For some reason, the parser broke the test script functionality.

Commit:
fixed leap.py to enable leap_test.py

Changes:
- Added if statements to load parser and execute program only if script
is called not imported

cmdellinger committed:
Fixed spelling

fixed spelling of "text" in line 48; previously said "test"

Commit:
moved import call to command-line implementation section

cmdellinger commented:
I updated the command-line to to use the docopt parser and refactored the leap function to eliminate nesting.

cc @barberj

barberj commented on python/leap/docopt.py:
```diff
@@ -0,0 +1,581 @@
+ * Licensed under terms of MIT license (see LICENSE-MIT)
+ * Copyright (c) 2013 Vladimir Keleshev, vladimir@keleshev.com
+
+"""
```
are you familiar with `pip`?

cmdellinger replied:
Yes, but for some reason docopt isn't playing well with my computer. I get "no module" errors or it returns __doc__ in python 2 and 3 respectively. I haven't been able to fix it.

cmdellinger replied:
I had to install/update setuptools and pip for python 2. I'll go through and delete all the docopt.py scripts.

Commit:
deleted docopt.py

deleted docopt.py from individual exercise folder

cmdellinger commented:
deleted docopt.py

cc @barberj

barberj commented:
Awesome I'll take a look tonight

Sent from my iPhone

> On Jan 17, 2017, at 7:28 AM, Charles M Dellinger <notifications@github.com> wrote:
> 
> deleted docopt.py
> 
> cc @barberj
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

cmdellinger committed:
Merge branch 'master' into python-leap
